deeppavlov/dataset_iterators/ner_few_shot_iterator.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from random import Random
+from typing import List, Dict, Tuple, Any, Iterator
+import numpy as np
+import re
+from copy import deepcopy
+from collections import Counter
+from itertools import chain
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.data_learning_iterator import DataLearningIterator
+
+
+@register('ner_few_shot_iterator')
+class NERFewShotIterator(DataLearningIterator):
+    """Dataset iterator for learning models, e. g. neural networks.
+
+    Args:
+        data: list of (x, y) pairs for every data type in ``'train'``, ``'valid'`` and ``'test'``
+        seed: random seed for data shuffling
+        shuffle: whether to shuffle data during batching
+
+    Attributes:
+        shuffle: whether to shuffle data during batching
+        random: instance of ``Random`` initialized with a seed
+    """
+    def split(self, *args, **kwargs):
+        pass
+
+    def __init__(self, data: Dict[str, List[Tuple[Any, Any]]],
+                 seed: int = None, shuffle: bool = True,
+                 target_tag: str = None,
+                 filter_bi: bool = True,
+                 n_train_samples: int = 20,
+                 remove_not_targets = True,
+                 *args, **kwargs) -> None:
+        self.shuffle = shuffle
+        self.target_tag = target_tag
+        self.filter_bi = filter_bi
+        self.n_train_samples = n_train_samples
+        self.remove_not_targets = remove_not_targets
+
+
+        if self.target_tag is None:
+            raise RuntimeError('You must provide a target tag to NERFewShotIterator!')
+
+        self.random = Random(seed)
```

Comment:
can't some of this code be replaced with `super().__init__(...)`?

Reply:
Done